PWGJE/Tasks/PhiInJetsJE.cxx

```diff
@@ -0,0 +1,412 @@
+      return false;
+    
+    if (std::abs(track.dcaXY()) > c_MaxDCArToPVcut)
+      return false;
+    
+    if (std::abs(track.dcaZ()) > c_MaxDCAzToPVcut)
+      return false;
+    
+    if (c_PrimaryTrack && !track.isPrimaryTrack())
+      return false;
+    
+    if(std::abs(track.tpcNClsFindable()) < c_nFindableTPCClusters)
+      return false;
+    
+    if(std::abs(track.tpcNClsCrossedRows()) < c_nTPCCrossedRows)
+      return false;
+    
+    if(std::abs(track.tpcCrossedRowsOverFindableCls()) > c_nRowsOverFindable)
+      return false;
+    
+    if(std::abs(track.tpcChi2NCl() > c_nTPCChi2))
+      return false;
+    
+    if(std::abs(track.itsChi2NCl() > c_nITSChi2))
+      return false;
+    
+    if (c_ConnectedToPV && !track.isPVContributor())
+      return false;    
+    return true;
+  };
+
+  template <typename T>
+  bool trackPID(const T& candidate)
+  {
+    bool tpcPIDPassed{false}, tofPIDPassed{false};
+    if (std::abs(candidate.tpcNSigmaKa()) < c_nTPCPID) 
+      tpcPIDPassed = true;
+    
+    if (candidate.hasTOF()) {
+      if (std::abs(candidate.tofNSigmaKa()) < c_nTOFPID) {
+        tofPIDPassed = true;
+      }
+    }
+    
+    else 
+      tofPIDPassed = true;
+    
+    if (tpcPIDPassed && tofPIDPassed) {
+      return true;
+    }
+    return false;
+  }
+
+  
+  template <bool IsMC, bool IsMix, typename TracksType, typename JetType>
+  void MinvReconstruction(double mult, const TracksType& trk1, const TracksType& trk2, const JetType& jets) {
+    TLorentzVector lDecayDaughter1, lDecayDaughter2, lResonance;
+
+    if(!trackSelection(trk1) || !trackSelection(trk2))
+      return;
+      
+    if(!trackPID(trk1) || !trackPID(trk2))
+      return;
+
+    if (trk1.index() == trk2.index())
+      return; // We need to run (0,1), (1,0) pairs as well. but same id pairs are not needed.
+    
+    lDecayDaughter1.SetXYZM(trk1.px(), trk1.py(), trk1.pz(), massKa);
+    lDecayDaughter2.SetXYZM(trk2.px(), trk2.py(), trk2.pz(), massKa);
+    lResonance = lDecayDaughter1 + lDecayDaughter2;
+
+    if (abs(lResonance.Rapidity()) > 0.5)
+      return;
+
+    if (trk1.sign() * trk2.sign() < 0) {
+      JEhistos.fill(HIST("hUSS_1D"), lResonance.M());
+	
+      if(lResonance.Pt() > 2.0 && lResonance.Pt() < 3)
+	JEhistos.fill(HIST("hUSS_1D_2_3"), lResonance.M());
+
+      JEhistos.fill(HIST("hUSS"), mult, lResonance.Pt(), lResonance.M());
+    }
+    else if(trk1.sign() * trk2.sign() > 0){
+	
+      JEhistos.fill(HIST("hLSS_1D"), lResonance.M());	
+      if(lResonance.Pt() > 2.0 && lResonance.Pt() < 3)
+	JEhistos.fill(HIST("hLSS_1D_2_3"), lResonance.M());
+
+      JEhistos.fill(HIST("hLSS"), mult, lResonance.Pt(), lResonance.M());	
+    }
+
+    bool jetFlag = false;
+    for (auto const& jet : jets){
+      
+      double phidiff = TVector2::Phi_mpi_pi(jet.phi()-lResonance.Phi());
+      double etadiff = jet.eta() - lResonance.Eta();     
+      double R = TMath::Sqrt((etadiff*etadiff) + (phidiff*phidiff));
```

Comment:
All this can be done with `RecoDecay`.

Reply:
I prefer to use these for now, as they are familiar to me. I will update them once I can confirm that my code works on hyperloop.